hoodie-client/src/main/java/com/uber/hoodie/table/HoodieTable.java

```diff
@@ -33,26 +32,31 @@
 import com.uber.hoodie.common.util.AvroUtils;
 import com.uber.hoodie.config.HoodieWriteConfig;
 import com.uber.hoodie.exception.HoodieException;
+import com.uber.hoodie.exception.HoodieIOException;
 import com.uber.hoodie.exception.HoodieSavepointException;
 import com.uber.hoodie.index.HoodieIndex;
+import com.uber.hoodie.io.ConsistencyCheck;
 import java.io.IOException;
 import java.io.Serializable;
 import java.util.Iterator;
 import java.util.List;
-import java.util.Optional;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 import org.apache.hadoop.conf.Configuration;
 import org.apache.spark.Partitioner;
 import org.apache.spark.api.java.JavaRDD;
 import org.apache.spark.api.java.JavaSparkContext;
-import scala.Tuple2;
 
 /**
  * Abstract implementation of a HoodieTable
  */
 public abstract class HoodieTable<T extends HoodieRecordPayload> implements Serializable {
 
+  // time between successive attempts to ensure written data's metadata is consistent on storage
+  private static long INITIAL_CONSISTENCY_CHECK_INTERVAL_MS = 1000L;
```

Comment:
Can these 2 depend on the efficiency of the cluster the user is running ? Some clusters may take longer time (for eventual consistency) while some may take shorter, is it good to have configs for these ?

Reply:
I want to punt on this for now. Exponential backoff should help allieviate the need for configs. Happy to discuss lower or higher thresholds though

Reply:
I have seen sometimes writes appearing after 1-2 minutes in S3. Since the common case is for the file to show up eventually, can we bump the higher threshold to a larger value - (5 - 10 minute) ?

Reply:
Okay I will bump up the number of attempts such that it will wait upto 5 mins before failing